### PR TITLE
email preferences

### PR DESCRIPTION
**Pull request by jowlee**
Email preferences added to user setting page. 
API endpoint created for updating preferences

### CODE REVIEW

File: client/app/account/settings/settings.html [hunk 1]
```diff
@@ -1,54 +1,79 @@
...
+        <div class="form-check">
+          <label class="form-check-label">
+            <input type="checkbox" class="form-check-input" ng-model="emailPreferences.unsubscribe">
+            Unsubscribe from emails
+          </label>
+        </div>
+        <h5>Event Reminder</h5>
+        <div ng-repeat='emailAhead in emailAheadOptions' class="form-check disabled">
+          <label class="form-check-label">
+            <input type="checkbox" class="form-check-input" ng-model="emailAhead.applied" ng-disabled="!emailPreferences.unsubscribe">
```
Comment (kmcnellis) [inline client/app/account/settings/settings.html]: This is confusing. The user should not have to click unsubscribe to change the event reminders. Instead they should be able to select any of them and click update (remove the Unsubscribe from emails checkbox).  There should also be a "Unsubscribe from all" button that deselects all and submits.

File: client/app/account/settings/settings.html [hunk 2]
```diff
@@ -1,54 +1,79 @@
...
+            {{ emailAhead.time }} before the event
```
Comment (kmcnellis) [inline client/app/account/settings/settings.html]: The text should have a click pointer to indicate that you can click on the text

**Review by kmcnellis — CHANGES_REQUESTED** [review]

**Comment by kmcnellis** [comment]
Where are the changes?

**Comment by vakrao** [comment]
I made a button that would enable and disable the ability to change the notifications. I had an un subscribe from all button, but, I thought it made more sense just to un-check all the options and then save. I can re-add the unsubscribe button if this setup does not work.

**Comment by kmcnellis** [comment]
I don't really like the change button. The combination with the save button makes it look like a huge chunk of purple. The caption should either be just "edit", and be a small non-colored button next to "Email Notification" header, or non colored and alone at the bottom. Once in edit mode, it should switch to cancel. The save button should only be present in the edit mode

**Comment by kmcnellis** [comment]
screen shot 2017-03-06 at 1 26 31 pm
screen shot 2017-03-06 at 1 26 09 pm

**Comment by vakrao** [comment]
Okay, will work on changing the look of this

**Comment by kmcnellis** [comment]
Thanks!

**Pull request merged**